feat: Add custom fields support for getIssues and countIssues tools

nix-tkobayashi opened the pull request:
## Description

This PR adds custom field filtering capability to the `getIssues` and `countIssues` tools.

## Changes

- Added `customFields` parameter to both tools' schemas
- Implemented custom field transformation using the existing `customFieldsToPayload` helper
- Added comprehensive test coverage for the new functionality

## Features

- Support for string, number, and array values in custom fields
- Maintains backward compatibility
- Follows Backlog API v2 specifications

## Testing

- Added 8 new test cases (4 for each tool)
- All existing tests continue to pass
- No linting errors

## Example Usage

```typescript
// Search for issues with specific custom field values
await mcp.getIssues({
  projectId: [100],
  customFields: [
    { id: 12345, value: "PROJECT-001" },
    { id: 67890, value: ["option1", "option2"] }
  ]
});
```

## Note

Due to Backlog API limitations, filtering by null custom field values is not supported directly. Users need to retrieve results and filter client-side for null values.

trknhr commented:
Sorry, I misoperated earlier. I've just made it public, so the comments should be visible now.
Could you check the PR again when you get a chance?

nix-tkobayashi commented:
Thank you for the review!

I've applied both suggestions:
- Changed the description from "Custom field filters" to "Custom fields" for consistency with other parameters
- Updated the handler functions to use destructuring directly in the parameters

I also applied the same changes to `getIssues.ts` to maintain consistency across the codebase, as it had the same pattern.

All tests are passing. Please let me know if there's anything else that needs adjustment!

trknhr commented:
Thank you
